From 49bf4f3a757b53d28718521d1e402f9afeeb6053 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Sun, 13 Sep 2026 17:49:54 +0200
Subject: [PATCH] Adaptation to new struct drm_atomic_commit, related to commit
 "drm: Rename struct drm_atomic_state to drm_atomic_commit" (Maxime Ripard, 27
 Apr 2026)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 kernel/conftest.sh                     | 75 ++++++++++++++++++--------
 kernel/nvidia-drm/nvidia-drm-crtc.c    | 12 ++---
 kernel/nvidia-drm/nvidia-drm-modeset.c | 38 ++++++-------
 kernel/nvidia-drm/nvidia-drm-modeset.h | 12 ++---
 kernel/nvidia-drm/nvidia-drm-priv.h    | 11 ++++
 kernel/nvidia-drm/nvidia-drm.Kbuild    |  6 +--
 6 files changed, 99 insertions(+), 55 deletions(-)

diff --git a/conftest.sh b/conftest.sh
index 36aa4ce..d4677b7 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -3282,9 +3282,9 @@ compile_test() {
             compile_check_conftest "$CODE" "NV_DRM_ATOMIC_MODESET_NONBLOCKING_COMMIT_AVAILABLE" "" "generic"
         ;;
 
-        drm_atomic_state_ref_counting)
+        drm_atomic_commit_ref_counting)
             #
-            # Determine if functions drm_atomic_state_get/put() are
+            # Determine if functions drm_atomic_commit_get/put() are
             # present.
             #
             # Added by commit 0853695c3ba4 ("drm: Add reference counting to
@@ -3294,11 +3294,18 @@ compile_test() {
             #if defined(NV_DRM_DRM_ATOMIC_H_PRESENT)
             #include <drm/drm_atomic.h>
             #endif
-            void conftest_drm_atomic_state_get(void) {
-                drm_atomic_state_get();
+            #include <linux/version.h>
+
+            // Rel. commit \"drm: Rename struct drm_atomic_state to drm_atomic_commit\" (Maxime Ripard, 27 Apr 2026)
+            #if LINUX_VERSION_CODE < KERNEL_VERSION(7, 2, 0)
+            #define drm_atomic_commit_get drm_atomic_state_get
+            #endif
+
+            void conftest_drm_atomic_commit_get(void) {
+                drm_atomic_commit_get();
             }"
 
-            compile_check_conftest "$CODE" "NV_DRM_ATOMIC_STATE_REF_COUNTING_PRESENT" "" "functions"
+            compile_check_conftest "$CODE" "NV_DRM_ATOMIC_COMMIT_REF_COUNTING_PRESENT" "" "functions"
         ;;
 
         vm_ops_fault_removed_vma_arg)
@@ -3866,8 +3873,15 @@ compile_test() {
             #
             CODE="
             #include <drm/drm_atomic_helper.h>
+            #include <linux/version.h>
+
+            // Rel. commit \"drm: Rename struct drm_atomic_state to drm_atomic_commit\" (Maxime Ripard, 27 Apr 2026)
+            #if LINUX_VERSION_CODE < KERNEL_VERSION(7, 2, 0)
+            #define drm_atomic_commit drm_atomic_state
+            #endif
+
             void conftest_drm_atomic_helper_swap_state_has_stall_arg(
-                    struct drm_atomic_state *state,
+                    struct drm_atomic_commit *state,
                     bool stall) {
                 (void)drm_atomic_helper_swap_state(state, stall);
             }"
@@ -3885,8 +3899,15 @@ compile_test() {
             #
             CODE="
             #include <drm/drm_atomic_helper.h>
+            #include <linux/version.h>
+
+            // Rel. commit \"drm: Rename struct drm_atomic_state to drm_atomic_commit\" (Maxime Ripard, 27 Apr 2026)
+            #if LINUX_VERSION_CODE < KERNEL_VERSION(7, 2, 0)
+            #define drm_atomic_commit drm_atomic_state
+            #endif
+
             int conftest_drm_atomic_helper_swap_state_return_int(
-                    struct drm_atomic_state *state,
+                    struct drm_atomic_commit *state,
                     bool stall) {
                 return drm_atomic_helper_swap_state(state, stall);
             }"
@@ -4340,10 +4361,10 @@ compile_test() {
             compile_check_conftest "$CODE" "NV_DRM_DRIVER_HAS_GEM_PRIME_CALLBACKS" "" "types"
         ;;
 
-        drm_crtc_atomic_check_has_atomic_state_arg)
+        drm_crtc_atomic_check_has_atomic_commit_arg)
             #
             # Determine if drm_crtc_helper_funcs::atomic_check takes 'state'
-            # argument of 'struct drm_atomic_state' type.
+            # argument of 'struct drm_atomic_commit' type.
             #
             # The commit 29b77ad7b9ca ("drm/atomic: Pass the full state to CRTC
             # atomic_check") passed the full atomic state to
@@ -4357,12 +4378,18 @@ compile_test() {
             #
             echo "$CONFTEST_PREAMBLE
             #include <drm/drm_modeset_helper_vtables.h>
+            #include <linux/version.h>
+
+            // Rel. commit \"drm: Rename struct drm_atomic_state to drm_atomic_commit\" (Maxime Ripard, 27 Apr 2026)
+            #if LINUX_VERSION_CODE < KERNEL_VERSION(7, 2, 0)
+            #define drm_atomic_commit drm_atomic_state
+            #endif
 
             static const struct drm_crtc_helper_funcs *funcs;
-            typeof(*funcs->atomic_check) conftest_drm_crtc_atomic_check_has_atomic_state_arg;
+            typeof(*funcs->atomic_check) conftest_drm_crtc_atomic_check_has_atomic_commit_arg;
 
-            int conftest_drm_crtc_atomic_check_has_atomic_state_arg(
-                    struct drm_crtc *crtc, struct drm_atomic_state *state) {
+            int conftest_drm_crtc_atomic_check_has_atomic_commit_arg(
+                    struct drm_crtc *crtc, struct drm_atomic_commit *state) {
                 return 0;
             }" > conftest$$.c
 
@@ -4371,9 +4398,9 @@ compile_test() {
 
             if [ -f conftest$$.o ]; then
                 rm -f conftest$$.o
-                echo "#define NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG" | append_conftest "types"
+                echo "#define NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG" | append_conftest "types"
             else
-                echo "#undef NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG" | append_conftest "types"
+                echo "#undef NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG" | append_conftest "types"
             fi
         ;;
 
@@ -4418,10 +4445,10 @@ compile_test() {
             echo "#undef NV_UNSAFE_FOLLOW_PFN_PRESENT" | append_conftest "functions"
         ;;
 
-        drm_plane_atomic_check_has_atomic_state_arg)
+        drm_plane_atomic_check_has_atomic_commit_arg)
             #
             # Determine if drm_plane_helper_funcs::atomic_check takes 'state'
-            # argument of 'struct drm_atomic_state' type.
+            # argument of 'struct drm_atomic_commit' type.
             #
             # The commit 7c11b99a8e58 ("drm/atomic: Pass the full state to
             # planes atomic_check") passed the full atomic state to
@@ -4435,12 +4462,18 @@ compile_test() {
             #
             echo "$CONFTEST_PREAMBLE
             #include <drm/drm_modeset_helper_vtables.h>
+            #include <linux/version.h>
+
+            // Rel. commit \"drm: Rename struct drm_atomic_state to drm_atomic_commit\" (Maxime Ripard, 27 Apr 2026)
+            #if LINUX_VERSION_CODE < KERNEL_VERSION(7, 2, 0)
+            #define drm_atomic_commit drm_atomic_state
+            #endif
 
             static const struct drm_plane_helper_funcs *funcs;
-            typeof(*funcs->atomic_check) conftest_drm_plane_atomic_check_has_atomic_state_arg;
+            typeof(*funcs->atomic_check) conftest_drm_plane_atomic_check_has_atomic_commit_arg;
 
-            int conftest_drm_plane_atomic_check_has_atomic_state_arg(
-                    struct drm_plane *plane, struct drm_atomic_state *state) {
+            int conftest_drm_plane_atomic_check_has_atomic_commit_arg(
+                    struct drm_plane *plane, struct drm_atomic_commit *state) {
                 return 0;
             }" > conftest$$.c
 
@@ -4449,9 +4482,9 @@ compile_test() {
 
             if [ -f conftest$$.o ]; then
                 rm -f conftest$$.o
-                echo "#define NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG" | append_conftest "types"
+                echo "#define NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG" | append_conftest "types"
             else
-                echo "#undef NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG" | append_conftest "types"
+                echo "#undef NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG" | append_conftest "types"
             fi
         ;;
 
diff --git a/nvidia-drm/nvidia-drm-crtc.c b/nvidia-drm/nvidia-drm-crtc.c
index a099cb1..de544c2 100644
--- a/nvidia-drm/nvidia-drm-crtc.c
+++ b/nvidia-drm/nvidia-drm-crtc.c
@@ -132,15 +132,15 @@ plane_req_config_update(struct drm_plane_state *plane_state,
         old_config.dstHeight != req_config->config.dstHeight;
 }
 
-#if defined(NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG)
+#if defined(NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG)
 static int nv_drm_plane_atomic_check(struct drm_plane *plane,
-                                     struct drm_atomic_state *state)
+                                     struct drm_atomic_commit *state)
 #else
 static int nv_drm_plane_atomic_check(struct drm_plane *plane,
                                      struct drm_plane_state *plane_state)
 #endif
 {
-#if defined(NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG)
+#if defined(NV_DRM_PLANE_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG)
     struct drm_plane_state *plane_state =
         drm_atomic_get_new_plane_state(state, plane);
 #endif
@@ -339,15 +339,15 @@ static int head_modeset_config_attach_connector(
  * the 'nv_drm_crtc_state::req_config', that is fine becase 'nv_drm_crtc_state'
  * will be discarded if ->atomic_check() fails.
  */
-#if defined(NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG)
+#if defined(NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG)
 static int nv_drm_crtc_atomic_check(struct drm_crtc *crtc,
-                                    struct drm_atomic_state *state)
+                                    struct drm_atomic_commit *state)
 #else
 static int nv_drm_crtc_atomic_check(struct drm_crtc *crtc,
                                     struct drm_crtc_state *crtc_state)
 #endif
 {
-#if defined(NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_STATE_ARG)
+#if defined(NV_DRM_CRTC_ATOMIC_CHECK_HAS_ATOMIC_COMMIT_ARG)
     struct drm_crtc_state *crtc_state =
         drm_atomic_get_new_crtc_state(state, crtc);
 #endif
diff --git a/nvidia-drm/nvidia-drm-modeset.c b/nvidia-drm/nvidia-drm-modeset.c
index b6ef1e7..f37c6bb 100644
--- a/nvidia-drm/nvidia-drm-modeset.c
+++ b/nvidia-drm/nvidia-drm-modeset.c
@@ -43,23 +43,23 @@
 #include <drm/drm_atomic_helper.h>
 #include <drm/drm_crtc.h>
 
-struct nv_drm_atomic_state {
+struct nv_drm_atomic_commit {
     struct NvKmsKapiRequestedModeSetConfig config;
-    struct drm_atomic_state base;
+    struct drm_atomic_commit base;
 };
 
-static inline struct nv_drm_atomic_state *to_nv_atomic_state(
-    struct drm_atomic_state *state)
+static inline struct nv_drm_atomic_commit *to_nv_atomic_commit(
+    struct drm_atomic_commit *state)
 {
-    return container_of(state, struct nv_drm_atomic_state, base);
+    return container_of(state, struct nv_drm_atomic_commit, base);
 }
 
-struct drm_atomic_state *nv_drm_atomic_state_alloc(struct drm_device *dev)
+struct drm_atomic_commit *nv_drm_atomic_state_alloc(struct drm_device *dev)
 {
-    struct nv_drm_atomic_state *nv_state =
+    struct nv_drm_atomic_commit *nv_state =
             nv_drm_calloc(1, sizeof(*nv_state));
 
-    if (nv_state == NULL || drm_atomic_state_init(dev, &nv_state->base) < 0) {
+    if (nv_state == NULL || drm_atomic_commit_init(dev, &nv_state->base) < 0) {
         nv_drm_free(nv_state);
         return NULL;
     }
@@ -67,16 +67,16 @@ struct drm_atomic_state *nv_drm_atomic_state_alloc(struct drm_device *dev)
     return &nv_state->base;
 }
 
-void nv_drm_atomic_state_clear(struct drm_atomic_state *state)
+void nv_drm_atomic_state_clear(struct drm_atomic_commit *state)
 {
-    drm_atomic_state_default_clear(state);
+    drm_atomic_commit_default_clear(state);
 }
 
-void nv_drm_atomic_state_free(struct drm_atomic_state *state)
+void nv_drm_atomic_state_free(struct drm_atomic_commit *state)
 {
-    struct nv_drm_atomic_state *nv_state =
-                    to_nv_atomic_state(state);
-    drm_atomic_state_default_release(state);
+    struct nv_drm_atomic_commit *nv_state =
+                    to_nv_atomic_commit(state);
+    drm_atomic_commit_default_release(state);
     nv_drm_free(nv_state);
 }
 
@@ -106,12 +106,12 @@ void nv_drm_atomic_state_free(struct drm_atomic_state *state)
  */
 static int
 nv_drm_atomic_apply_modeset_config(struct drm_device *dev,
-                                   struct drm_atomic_state *state,
+                                   struct drm_atomic_commit *state,
                                    bool commit)
 {
     struct nv_drm_device *nv_dev = to_nv_device(dev);
     struct NvKmsKapiRequestedModeSetConfig *requested_config =
-        &(to_nv_atomic_state(state)->config);
+        &(to_nv_atomic_commit(state)->config);
     struct drm_crtc *crtc;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
     struct drm_crtc_state *old_crtc_state, *new_crtc_state;
@@ -180,7 +180,7 @@ nv_drm_atomic_apply_modeset_config(struct drm_device *dev,
 }
 
 int nv_drm_atomic_check(struct drm_device *dev,
-                        struct drm_atomic_state *state)
+                        struct drm_atomic_commit *state)
 {
     int ret = 0;
 
@@ -224,7 +224,7 @@ static void __nv_drm_handle_flip_event(struct nv_drm_crtc *nv_crtc)
 }
 
 int nv_drm_atomic_commit(struct drm_device *dev,
-                         struct drm_atomic_state *state,
+                         struct drm_atomic_commit *state,
                          bool nonblock)
 {
     int ret = -EBUSY;
@@ -364,7 +364,7 @@ int nv_drm_atomic_commit(struct drm_device *dev,
         }
     }
 
-#if defined(NV_DRM_ATOMIC_STATE_REF_COUNTING_PRESENT)
+#if defined(NV_DRM_ATOMIC_COMMIT_REF_COUNTING_PRESENT)
     /*
      * If ref counting is present, state will be freed when the caller
      * drops its reference after we return.
diff --git a/nvidia-drm/nvidia-drm-modeset.h b/nvidia-drm/nvidia-drm-modeset.h
index effb990..418891e 100644
--- a/nvidia-drm/nvidia-drm-modeset.h
+++ b/nvidia-drm/nvidia-drm-modeset.h
@@ -30,17 +30,17 @@
 #include "nvkms-kapi.h"
 
 struct drm_device;
-struct drm_atomic_state;
+struct drm_atomic_commit;
 
-struct drm_atomic_state *nv_drm_atomic_state_alloc(struct drm_device *dev);
-void nv_drm_atomic_state_clear(struct drm_atomic_state *state);
-void nv_drm_atomic_state_free(struct drm_atomic_state *state);
+struct drm_atomic_commit *nv_drm_atomic_state_alloc(struct drm_device *dev);
+void nv_drm_atomic_state_clear(struct drm_atomic_commit *state);
+void nv_drm_atomic_state_free(struct drm_atomic_commit *state);
 
 int nv_drm_atomic_check(struct drm_device *dev,
-                        struct drm_atomic_state *state);
+                        struct drm_atomic_commit *state);
 
 int nv_drm_atomic_commit(struct drm_device *dev,
-                         struct drm_atomic_state *state, bool nonblock);
+                         struct drm_atomic_commit *state, bool nonblock);
 
 
 void nv_drm_handle_flip_occurred(struct nv_drm_device *nv_dev,
diff --git a/nvidia-drm/nvidia-drm-priv.h b/nvidia-drm/nvidia-drm-priv.h
index 953c2eb..df7db28 100644
--- a/nvidia-drm/nvidia-drm-priv.h
+++ b/nvidia-drm/nvidia-drm-priv.h
@@ -25,6 +25,17 @@
 
 #include "nvidia-drm-conftest.h" /* NV_DRM_AVAILABLE */
 
+#include <linux/version.h>
+
+// Rel. commit "drm: Rename struct drm_atomic_state to drm_atomic_commit" (Maxime Ripard, 27 Apr 2026)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(7, 2, 0)
+#define drm_atomic_commit                 drm_atomic_state
+#define drm_atomic_commit_init            drm_atomic_state_init
+#define drm_atomic_commit_free            drm_atomic_state_free
+#define drm_atomic_commit_default_clear   drm_atomic_state_default_clear
+#define drm_atomic_commit_default_release drm_atomic_state_default_release
+#endif
+
 #if defined(NV_DRM_AVAILABLE)
 
 #if defined(NV_DRM_DRMP_H_PRESENT)
diff --git a/nvidia-drm/nvidia-drm.Kbuild b/nvidia-drm/nvidia-drm.Kbuild
index cb49130..2c3decd 100644
--- a/nvidia-drm/nvidia-drm.Kbuild
+++ b/nvidia-drm/nvidia-drm.Kbuild
@@ -58,7 +58,7 @@ NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_reinit_primary_mode_group
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += get_user_pages_remote
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += get_user_pages
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_gem_object_lookup
-NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_atomic_state_ref_counting
+NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_atomic_commit_ref_counting
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_driver_has_gem_prime_res_obj
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_atomic_helper_connector_dpms
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += drm_connector_funcs_have_mode_in_name
@@ -100,9 +100,9 @@ NV_CONFTEST_TYPE_COMPILE_TESTS += drm_driver_master_set_has_int_return_type
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_driver_has_gem_free_object
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_prime_pages_to_sg_has_drm_device_arg
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_driver_has_gem_prime_callbacks
-NV_CONFTEST_TYPE_COMPILE_TESTS += drm_crtc_atomic_check_has_atomic_state_arg
+NV_CONFTEST_TYPE_COMPILE_TESTS += drm_crtc_atomic_check_has_atomic_commit_arg
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_gem_object_vmap_has_map_arg
-NV_CONFTEST_TYPE_COMPILE_TESTS += drm_plane_atomic_check_has_atomic_state_arg
+NV_CONFTEST_TYPE_COMPILE_TESTS += drm_plane_atomic_check_has_atomic_commit_arg
 NV_CONFTEST_TYPE_COMPILE_TESTS += drm_device_has_pdev
 NV_CONFTEST_TYPE_COMPILE_TESTS += dma_resv_add_fence
 NV_CONFTEST_TYPE_COMPILE_TESTS += dma_resv_reserve_fences
-- 
2.55.0

